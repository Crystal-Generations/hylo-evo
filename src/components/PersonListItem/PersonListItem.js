import React from 'react'
import cx from 'classnames'

import RoundImage from 'components/RoundImage'
import './PersonListItem.scss'

const { any, bool, func, shape, string } = React.PropTypes

const personType = shape({
  id: any,
  name: string,
  avatarUrl: string,
  community: string
})

export default function PersonListItem ({ addParticipant, active, person }) {
  return <li styleName={cx('person-list-item', { active })}>
    <RoundImage url={person.avatarUrl} styleName='avatar' />
    <span styleName='name'>{person.name}</span>
    <span styleName='community'>{person.community}</span>
  </li>
}

PersonListItem.propTypes = {
  active: bool,
  addParticipant: func,
  person: personType
}
