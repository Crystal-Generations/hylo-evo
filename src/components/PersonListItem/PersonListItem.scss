.person-list-item {
  list-style: none;
  display: flex;
  align-items: center;
  justify-content: center;
  max-width: 500px;
  margin: $space-1x 0;
}

.active {
  background-color: $color-rhino-30;
}

.avatar {
  margin-right: $space-4x;
}

.name {
  composes: bdy-lt-sm from 'css/typography.scss';
  flex: 1;
  padding-right: $space-2x;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.community {
  composes: faint-lt-sm from 'css/typography.scss';
}
