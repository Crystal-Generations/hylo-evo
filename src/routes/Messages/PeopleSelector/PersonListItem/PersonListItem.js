import PropTypes from 'prop-types'
import React from 'react'
import cx from 'classnames'

import RoundImage from 'components/RoundImage'
import './PersonListItem.scss'

export default function PersonListItem ({ active, onClick, onMouseOver, person }) {
  return <li styleName={cx('person-list-item', { active })} onClick={onClick} onMouseOver={onMouseOver}>
    <RoundImage url={person.avatarUrl} styleName='avatar' medium />
    <span styleName='name'>{person.name}</span>
    <span styleName='community'>{person.community}</span>
  </li>
}

const personType = PropTypes.shape({
  id: PropTypes.any,
  name: PropTypes.string,
  avatarUrl: PropTypes.string,
  community: PropTypes.string
})

PersonListItem.propTypes = {
  active: PropTypes.bool,
  onClick: PropTypes.func,
  onMouseOver: PropTypes.func,
  person: personType
}
