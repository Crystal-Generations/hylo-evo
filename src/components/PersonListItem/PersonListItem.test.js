import PersonListItem from './PersonListItem'
import { shallow } from 'enzyme'
import React from 'react'

it('matches the last snapshot', () => {
  const wrapper = shallow(<PersonListItem />)
  expect(wrapper).toMatchSnapshot()
})
