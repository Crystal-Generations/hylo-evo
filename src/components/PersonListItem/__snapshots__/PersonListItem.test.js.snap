// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the last snapshot 1`] = `
<li
  data-styleName="person-list-item active"
>
  Wombat
</li>
`;
